### ISSUE TITLE
Make the "stressing network" automatic ban less trigger-happy

### ISSUE THREAD

**Issue by Dune-jr** [teeworlds/teeworlds#1631]
I experience getting banned regularly when restarting the client for a setting to take effect. Would it be possible to make this condition more gentle?

**Comment by nheir** [comment]
This is a specific situation where you close somewhat the connection less than 1s after connecting.
Maybe add an debug option to the server not to ban is better than being more gentle.

**Comment by Dune-jr** [comment]
@nheir Double tapping enter does that. Wouldn't the server allowing one "stressful" connection before banning the IP be reasonable?

**Comment by Dune-jr** [comment]
> What is double tapping ? Connect followed by cancel ?

Yeah. Usually I want to skip the menus, but if I tap enter once too much I get banned on the server.

I mean, we could also **not** change the server but instead change the client to not disconnect too quickly, that solves it too.

**Comment by oy** [comment]
It's a protection from the join/leave spammers - should stay that way.
The client could prevent you from reconnecting yeah.

**Comment by nheir** [comment]
> > What is double tapping ? Connect followed by cancel ?
> 
> Yeah. Usually I want to skip the menus, but if I tap enter once too much I get banned on the server.
> 
> I mean, we could also **not** change the server but instead change the client to not disconnect too quickly, that solves it too.

It make sense, but I don't know where we should put this.

https://github.com/teeworlds/teeworlds/blob/2c6c9ea3e9ad87937ae3ff3a2fc880d2b0e744ad/src/game/client/components/menus.cpp#L1875-L1879
https://github.com/teeworlds/teeworlds/blob/2c6c9ea3e9ad87937ae3ff3a2fc880d2b0e744ad/src/engine/client/client.cpp#L570-L573

**Comment by heinrich5991** [comment]
Having a better protection against join-/leave-spammers might also be a good option. Currently, the best version for a custom client would be to never send the disconnect message if it recently connected, ruling out any "connection stress" bans. Thus, our current protection isn't really the best way to solve it.

**Comment by oy** [comment]
@heinrich5991 
Yeah, that should be improved - probably split that to a different issue?

**Comment by oy** [comment]
#1303 seems like sth else. There you need multiple IPs. What @heinrich5991 mentioned just needs one. You just time out without disconnecting so this "stressing network" automatic ban isn't triggered.

**Comment by jxsl13** [comment]
I am getting like at least one of these false positive bans per hour, which is bad, especially from a psychological point of view.
